graphics/drivers/framebuffer.s

```diff
@@ -382,21 +384,105 @@ fill_y:	sta VERA_DATA0
 	bne fill_y
 	rts
 
-; XXX TODO support other step sizes
 fill_pixels_with_step:
-	pha
+
+	; Is the step size equal to 320?
+	cpy #(320 >> 8)
+	bne fill_pixels_non_accelerated
+	cpx #(320 - 256)
+	bne fill_pixels_non_accelerated
+
+	; Step size is 320, fill using HW accelerated increment setting in VERA
+
+	; Temporarily set ADDR_H to use increment 14 (320)
+	tay
 	lda VERA_ADDR_H
-	ora #$70        ; increment in steps of $40
+	and #$01
+	ora #$E0
 	sta VERA_ADDR_H
-	pla
+	tya
+
+	jsr fill_pixels_hw_accelerated
+
+fill_pixels_reset_increment_and_rts:
+	; Restore ADDR_H to use increment 1
+	lda VERA_ADDR_H
+	and #$01
+	ora #$10
+	sta VERA_ADDR_H
+	rts
+
+fill_pixels_non_accelerated:
+
+	tay
+
+	; temporarily set increment to 0
+	lda #$FE
+	trb VERA_ADDR_H
+
 	ldx r0L
-@loop:	sta VERA_DATA0
-	inc VERA_ADDR_M ; increment hi -> add $140 = 320
-	bne :+
+	inx
+	inc r0H
```

Comment:
I think this `inc` was the only thing that puzzled me for a moment, but I think I see why it's here: to allow a `dec` to reach zero later.  Everything else I was able to follow just fine.

Reply:
Indeed. The reason is that dec does not have carry flag output to evaluate if we are done, it only have zero flag and negative output. Thus, to "emulate" a carry output, I increment it and wait for zero instead. I did the same thing for r0L, which is placed in X to improve speed.